### PR TITLE
fix(Core/Unit): Petrified Lichen Guard effect

### LINKED ISSUE #14415: Petrified Lichen Gaurd proc incorrect
Original CC report
https://github.com/chromiecraft/chromiecraft/issues/4575 confirmed on my local. 

### What client do you play on?

enGB

### Faction

Both

### Content Phase:

61-64

### Current Behaviour

[Petrified Lichen Guard](https://wowgaming.altervista.org/aowow/?item=25828) Casts lightning shield doing a static 13 damage to mobs. (spell ID 26364)
image

[broken shield.webm](https://user-images.githubusercontent.com/62584589/208229477-37f9afdf-2f39-4c81-bf3c-30b94374229a.webm)

### Expected Blizzlike Behaviour

The shield is not suppose to do static damage, but instead apply a debuff to the target. This can stack up to 5 times, increasing the dmg per tick with each stack. 

[Spore Cloud](https://wowgaming.altervista.org/aowow/?spell=32643) is the correct spell it should be casting.

### Source

Shield : https://wowgaming.altervista.org/aowow/?item=25828
Proc : https://wowgaming.altervista.org/aowow/?spell=32642
Applied Aura : https://wowgaming.altervista.org/aowow/?spell=32643

### Steps to reproduce the problem

1. be class that has a shield skill minimum level 62
2. .mod rep 970 honored
3. .additem 25828
4. Let an enemy attack you and watch output dmg and combat log

### Extra Notes

The currently used incorrect spell is overriding shamans spell shields - Water Shield and Lightning Shield, which makes this shield unusable by shamans due to having to reapply class spell shield again, which in turn overrides items one.  This bug makes shield unusable in pvp and pve alike for shaman.

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/790f444ffac09bfb9193fbd18aa7c2d730a02600

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-bg-item-reward](https://github.com/azerothcore/mod-bg-item-reward)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-cta-switch](https://github.com/azerothcore/mod-cta-switch)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna](https://github.com/azerothcore/mod-eluna)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-low-level-rbg](https://github.com/azerothcore/mod-low-level-rbg)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-progression-system](https://github.com/azerothcore/mod-progression-system)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-rdf-expansion](https://github.com/azerothcore/mod-rdf-expansion)
- [mod-transmog](https://github.com/azerothcore/mod-transmog)
- [mod-weekend-xp](https://github.com/azerothcore/mod-weekend-xp)
- [mod-instanced-worldbosses](https://github.com/nyeriah/mod-instanced-worldbosses)
- [mod-zone-difficulty](https://github.com/azerothcore/mod-zone-difficulty)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-exchange-npc](https://github.com/55Honey/Acore_ExchangeNpc)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by NinjaPleezAC**
Petrified Lichen Guard's aura was being applied and procced incorrectly. The aura should no longer overwrite other elemental shields. The aura should now correctly proc its stacking DoT effect.

Closes AzerothCore issue #14415

## Changes Proposed:
-  Resolved false positive in Lightning/Elemental shield conditionals in Unit::HandleProcTriggerSpell() and SpellInfo::LoadSpellSpecific()
-  These were incorrectly causing the Spore Cloud aura from Petrified Lichen Guard to proc the Lightning Shield on-hit effect, and conflict with other elemental auras.

## Issues Addressed:

- Closes #14415
- Closes https://github.com/chromiecraft/chromiecraft/issues/4575

## SOURCE:

https://www.wowhead.com/wotlk/item=25828/petrified-lichen-guard
https://www.wowhead.com/wotlk/spell=32642/spore-cloud

## Tests Performed:

- Built without errors
- Tested on Windows 10 x86_64
- Tested Petrified Lichen Guard on-block effect correctly applied stacking DoT
- Tested Lightning/Earth/Water Shield spells not being overwritten by equipping Petrified Lichen Guard
- Regression tested Lightning/Earth/Water Shield spells still correctly overwriting each other when cast
- Regression tested Lightning/Earth/Water Shield spells still correctly applying their effects

## How to Test the Changes:

1. Create/select a character that can equip shields, and is level 62+
2. .mod rep 970 honored
3. .additem 25828
4. Equip Petrified Lichen Guard and allow mobs to hit you
5. Observe correct application of DoT effect
6. Cast Lightning/Earth/Water Shield spells
7. Observe that the elemental shield spells overwrite each other, but not Petrified Lichen Guard's aura
8. Observe that Lightning/Earth/Water Shield spells still correctly proc their effects

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by BeefSteaker** [comment]
Tested, works as expected for me. 
Tested shaman shields and no issues/regressions

File: src/server/game/Entities/Unit/Unit.cpp [hunk 1]
```diff
@@ -9279,7 +9279,7 @@ bool Unit::HandleProcTriggerSpell(Unit* victim, uint32 damage, AuraEffect* trigg
             case SPELLFAMILY_SHAMAN:
                 {
                     // Lightning Shield (overwrite non existing triggered spell call in spell.dbc
-                    if (auraSpellInfo->SpellFamilyFlags[0] & 0x400)
+                    if (auraSpellInfo->SpellFamilyFlags[0] & 0x400 && auraSpellInfo->AttributesEx & 0x400)
```
Comment (Kitzunu) [inline src/server/game/Entities/Unit/Unit.cpp:9282]: Do not use the hex code for Attributes, instead use the defined enum for it.

Defined in SharedDefines.h

Also you can use HasAttribute()

File: src/server/game/Spells/SpellInfo.cpp [hunk 2]
```diff
@@ -2253,7 +2253,7 @@ SpellSpecificType SpellInfo::LoadSpellSpecific() const
             {
                 // family flags 10 (Lightning), 42 (Earth), 37 (Water), proc shield from T2 8 pieces bonus
                 if (SpellFamilyFlags[1] & 0x420
-                        || SpellFamilyFlags[0] & 0x00000400
+                        || (SpellFamilyFlags[0] & 0x00000400 && AttributesEx & 0x00000400)
```
Comment (Kitzunu) [inline src/server/game/Spells/SpellInfo.cpp:2256]: same as above

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Review by IntelligentQuantum — APPROVED** [review]

**Comment by NinjaSoftworks** [comment]
@Kitzunu are you available to re-review? I think merge is hung up on the requested changes.

**Comment by heyitsbench** [comment]
> I think merge is hung up on the requested changes.

It isn't, merges are done manually.

**Review by Kitzunu — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thanks for the pr.

**Pull request merged**